src/coreclr/src/tools/Common/Internal/Runtime/ReadyToRunInstructionSet.cs

```diff
@@ -0,0 +1,131 @@
+
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+// DO NOT EDIT THIS FILE! IT IS AUTOGENERATED
```

Comment:
In many projects I worked on, people refrained from pushing auto-generated code if it could be generated during build time. This avoids bitrotting the generators and people forgetting to run the script to update these files (the build system would know the dependencies and rebuild these `.cs` files when their source `.txt` are changed.)

Is checking in generated files common in .NET? (If not, is this because IDEs might not like files that are generated during build time?)

Reply:
There are a lot of benefits to checking in the files even when they can be auto-generated, IMO.

One example is being able to browse, index, and link to the code on GitHub and other sites without having to integrate with the CI system.

It also reduces the build time, ensures we know what code is being included in the compilation, and integrates better with IDEs (as some, as you indicated, do not play nicely with generated files).